### ISSUE TITLE
Compatibility issue:  ERRO 906 compat: self-check #3: dns integration check failed?

### ISSUE THREAD

**Issue by dazu89** [safing/portmaster#1007]
Intermittently, Portmaster reports a compatibility issue hinting at a VPN which I am not using.  I would be thankful for assistance in localising the issue.

**Debug Information**:
<details>
<summary>Version 1.0.0</summary>

```
Portmaster
version 1.0.0

commit tags/v1.0.0-0-g6868333425ff0c4ab25bba19e22f560da6eaced0
built with go1.19 (gc) windows/amd64
  using options main.go
  by user@docker
  on 18.10.2022

Licensed under the AGPLv3 license.
The source code is available here: https://github.com/safing/portmaster
```

</details>

<details>
<summary>Platform: Microsoft Windows 11 Pro 10.0.22621 Build 22621</summary>

```
System: Microsoft Windows 11 Pro windows (Standalone Workstation) 10.0.22621 Build 22621
Kernel: 10.0.22621 Build 22621 x86_64

```

</details>

<details>
<summary>Status: Trusted</summary>

```
ActiveSecurityLevel:   Trusted
SelectedSecurityLevel: Off
ThreatMitigationLevel: Trusted
CaptivePortal:         
OnlineStatus:          Online
```

</details>

<details>
<summary>Config: 7</summary>

```
core/enableNetworkRating: true
core/expertiseLevel: expert
filter/blockInternet: 4
filter/blockLAN: 4
filter/blockLocal: 4
filter/blockP2P: 6
filter/lists: [TRAC MAL DECEP BAD UNBREAK]
```

</details>

<details>
<summary>Resolvers: 7/7</summary>

```
Cloudflare (dot://cloudflare-dns.com:853#config)
  dot://cloudflare-dns.com:853#config
  Failing: false

Cloudflare (dot://cloudflare-dns.com:853#config)
  dot://cloudflare-dns.com:853#config
  Failing: false

2a01:8b80:1013:1::140 (dns://2a01:8b80:1013:1::140:53#system)
  dns://2a01:8b80:1013:1::140:53#system
  Failing: false

2a01:8b80:1013:1::139 (dns://2a01:8b80:1013:1::139:53#system)
  dns://2a01:8b80:1013:1::139:53#system
  Failing: false

185.42.245.140 (dns://185.42.245.140:53#system)
  dns://185.42.245.140:53#system
  Failing: false

185.42.245.139 (dns://185.42.245.139:53#system)
  dns://185.42.245.139:53#system
  Failing: false

185.42.245.135 (dns://185.42.245.135:53#system)
  dns://185.42.245.135:53#system
  Failing: false
```

</details>

<details>
<summary>SPN: disabled (module disabled)</summary>

```
HomeHubID:    
HomeHubName:  
HomeHubIP:    
Transport:    
---
Client:       true
PublicHub:    false
HubHasIPv4:   false
HubHasIPv6:   false
```

</details>

<details>
<summary>Compatibility: WFP State (374)</summary>

```
Edge traversal Teredo Authorization Sublayer   SubLayer   Edge traversal Teredo Authorization Sublayer                                 {7b6b11f6-cbb5-433c-ae06-6a4f0076e49e}
PortmasterInboundV4Callout                     Callout    This callout is used by the Portmaster to intercept inbound IPv4 traffic.    {05c55149-4732-4857-8d10-f178f3a06f8c}   [no provider key]   FWPM_LAYER_INBOUND_IPPACKET_V4
PortmasterInboundV4Filter                      Filter     This filter is used by the Portmaster to intercept inbound IPv4 traffic.     {7c329a28-af53-464c-9b8a-b3e8d5361ed5}   [no provider key]   FWPM_LAYER_INBOUND_IPPACKET_V4   {a87fb472-fc68-4805-8559-c6ae774773e0}
PortmasterInboundV6Callout                     Callout    This callout is used by the Portmaster to intercept inbound IPv6 traffic.    {ceff1df7-2baa-44c5-a6e5-73a95849bcff}   [no provider key]   FWPM_LAYER_INBOUND_IPPACKET_V6
PortmasterInboundV6Filter                      Filter     This filter is used by the Portmaster to intercept inbound IPv6 traffic.     {2cffa16e-3556-497b-804b-09ecc25ec0df}   [no provider key]   FWPM_LAYER_INBOUND_IPPACKET_V6   {a87fb472-fc68-4805-8559-c6ae774773e0}
PortmasterOutboundV4Callout                    Callout    This callout is used by the Portmaster to intercept outbound IPv4 traffic.   {41162b9e-8473-4b88-a5eb-04cf1d276b06}   [no provider key]   FWPM_LAYER_OUTBOUND_IPPACKET_V4
PortmasterOutboundV4Filter                     Filter     This filter is used by the Portmaster to intercept outbound IPv4 traffic.    {369acbb5-658c-4e58-86cf-4c46ad556fd0}   [no provider key]   FWPM_LAYER_OUTBOUND_IPPACKET_V4   {a87fb472-fc68-4805-8559-c6ae774773e0}
PortmasterOutboundV6Callout                    Callout    This callout is used by the Portmaster to intercept outbound IPv6 traffic.   {32bad112-6af4-4109-809b-c07570ba01b4}   [no provider key]   FWPM_LAYER_OUTBOUND_IPPACKET_V6
PortmasterOutboundV6Filter                     Filter     This filter is used by the Portmaster to intercept outbound IPv6 traffic.    {64856cc8-cc2a-4939-9ccb-349c5a820ceb}   [no provider key]   FWPM_LAYER_OUTBOUND_IPPACKET_V6   {a87fb472-fc68-4805-8559-c6ae774773e0}
PortmasterSublayer                             SubLayer   The Portmaster sublayer holds all it's filters.                              {a87fb472-fc68-4805-8559-c6ae774773e0}
Teredo socket option opt out block filter      Filter     [no description]                                                             {5a5ee689-84b9-4cf2-9fc8-244e0ab7e0a4}   {c0e4955a-27f8-4c07-ac1c-127e488d5d9f}   FWPM_LAYER_ALE_AUTH_RECV_ACCEPT_V6        {7b6b11f6-cbb5-433c-ae06-6a4f0076e49e}
```

</details>

<details>
<summary>Compat Module Error</summary>

```
Message: self-check #3: dns integration check failed: failed to receive test response: lookup a3c94898acb174c9f6cef16b012c0bbc7b.self-check.portmaster.home.arpa.: no such host
Timestamp: 2022-11-16 23:13:04.0783413 +0100 CET m=+26082.202530901
ModuleName: compat
TaskName: selfcheck
TaskType: 
Severity: error
PanicValue: %!s(<nil>)
StackTrace:

goroutine 100164 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/safing/portbase/modules.(*Module).NewErrorMessage(0xc0002a8780, {0x1692b4f, 0x9}, {0x181a0c0?, 0xc001b223a0?})
	/home/user/git/safing/portbase/modules/error.go:48 +0x49
github.com/safing/portmaster/compat.(*systemIssue).notify(0x1ddd640, {0x181a0c0, 0xc001b223a0})
	/home/user/git/safing/portmaster/compat/notify.go:144 +0x22b
github.com/safing/portmaster/compat.selfcheckTaskFunc({0x181fcd8?, 0xc001834a80?}, 0x16f58c9a5860?)
	/home/user/git/safing/portmaster/compat/module.go:110 +0x187
github.com/safing/portbase/modules.(*Task).executeWithLocking(0xc00050f4d0)
	/home/user/git/safing/portbase/modules/tasks.go:385 +0xbb
created by github.com/safing/portbase/modules.(*Task).runWithLocking
	/home/user/git/safing/portbase/modules/tasks.go:331 +0x305


```

</details>

<details>
<summary>Unexpected Logs</summary>

```
221116 23:05:41.453 pat/module:108 > ERRO 888 compat: self-check #3: dns integration check failed: failed to receive test response: lookup acaf4478c88c7af45f121c7b951b97031b.self-check.portmaster.home.arpa.: no such host
221116 23:05:46.655 pat/module:108 > ERRO 890 compat: self-check #3: dns integration check failed: failed to receive test response: lookup ad1d9b33789250eab0a1dcc501eb50f0eb.self-check.portmaster.home.arpa.: no such host
221116 23:05:52.022 pat/module:108 > ERRO 892 compat: self-check #3: dns integration check failed: failed to receive test response: lookup af48a445f4367bb14d99ae37958e5e184b.self-check.portmaster.home.arpa.: no such host
221116 23:05:57.417 pat/module:108 > ERRO 894 compat: self-check #3: dns integration check failed: failed to receive test response: lookup ab67fa7a7ceb69801f561bac3427df6f7b.self-check.portmaster.home.arpa.: no such host
221116 23:06:02.813 pat/module:108 > ERRO 896 compat: self-check #3: dns integration check failed: failed to receive test response: lookup a0a415657a1332b63dae14d405de0d15cb.self-check.portmaster.home.arpa.: no such host
221116 23:06:08.198 pat/module:108 > ERRO 898 compat: self-check #3: dns integration check failed: failed to receive test response: lookup aa47d2516018a0c2abfbf625082b1e434b.self-check.portmaster.home.arpa.: no such host
221116 23:06:13.595 pat/module:108 > ERRO 900 compat: self-check #3: dns integration check failed: failed to receive test response: lookup a4dd676bd56defc86e0da2811618ccbf0b.self-check.portmaster.home.arpa.: no such host
221116 23:06:19.004 pat/module:108 > ERRO 902 compat: self-check #3: dns integration check failed: failed to receive test response: lookup a690f639eb149fba3c31585d3e8288528b.self-check.portmaster.home.arpa.: no such host
221116 23:06:30.746 pat/module:108 > ERRO 904 compat: self-check #3: dns integration check failed: failed to receive test response: lookup a3ce3d686731195707f354a2e9d1490a2b.self-check.portmaster.home.arpa.: no such host
221116 23:06:36.790 pat/module:108 > ERRO 906 compat: self-check #3: dns integration check failed: failed to receive test response: lookup a075874c3412e4a61a1530cdbad011752b.self-check.portmaster.home.arpa.: no such host
221116 23:13:04.078 CURRENT TIME
```

</details>

**Comment by vlabo** [comment]
Thank you for the report. I will investigate the problem when I find some more time.
Just few questions to get more insight of the issue.

Does this happen often?  (on every dns integration check)
Do you have anything special with your setup that you think may interfere?

**Comment by 3941** [comment]
Happens here, too. Pretty much at least once a day. Logs look the same.

Takes maybe 10 minutes to self correct it. Maybe that's when another DNS check is fired.

No VPNs or anything else in use. I used to have Adguard installed, and toggling that would fix the Portmaster security status pretty much instantly, but I've since removed it and it still keeps happening.

**Comment by 3941** [comment]
I've got an email from you guys, just to thank me for posting here, so I guess I'll write another update, because things have changed. 

I've updated to the beta version a few days ago (not to try and fix THIS problem, but try to see if it helped with another. Specifically if clicking an app in the connection alert thing and choosing "change default" would actually take me to the proper page of the app settings to change the default action. Sadly it doesn't. It still points to the connection log of the app). 

Anyway, it seems to have fixed THIS issue, though. I don't think I've seen the popup since then.
